cpp-package/src/data_filter.cpp

```diff
@@ -214,35 +194,51 @@ std::pair<double *, double *> DataFilter::get_log_psd_welch (
     }
     double *ampl = new double[nfft / 2 + 1];
     double *freq = new double[nfft / 2 + 1];
-    int res =
-        ::get_log_psd_welch (data, data_len, nfft, overlap, sampling_rate, window, ampl, freq);
+    int res = ::get_psd_welch (data, data_len, nfft, overlap, sampling_rate, window, ampl, freq);
     if (res != (int)BrainFlowExitCodes::STATUS_OK)
     {
         delete[] ampl;
         delete[] freq;
-        throw BrainFlowException ("failed to get_log_psd_welch", res);
+        throw BrainFlowException ("failed to get_psd_welch", res);
     }
     return std::make_pair (ampl, freq);
 }
 
-std::pair<double *, double *> DataFilter::get_psd_welch (
-    double *data, int data_len, int nfft, int overlap, int sampling_rate, int window)
+std::pair<double *, double *> DataFilter::get_avg_band_powers (
+    double **data, int cols, int *channels, int channels_len, int sampling_rate, bool apply_filters)
 {
-    if ((nfft & (nfft - 1)) || (nfft <= 0))
+    if ((data == NULL) || (channels == NULL) || (channels_len < 1))
     {
         throw BrainFlowException (
-            "nfft is not power of 2", (int)BrainFlowExitCodes::INVALID_ARGUMENTS_ERROR);
+            "Invalid params", (int)BrainFlowExitCodes::INVALID_ARGUMENTS_ERROR);
     }
-    double *ampl = new double[nfft / 2 + 1];
-    double *freq = new double[nfft / 2 + 1];
-    int res = ::get_psd_welch (data, data_len, nfft, overlap, sampling_rate, window, ampl, freq);
+    double *data_1d = new double[cols * channels_len];
+    double *avg_bands = new double[5];
+    double *stddev_bands = new double[5];
```

Comment:
no reason for these to be dyamically allocated

Reply:
its returned to the user and you can not return a pointer to local variable

Reply:
ah, that's right